### PR TITLE
add number of partitions as aggregator settings

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: core-plugins/docs/Distinct-batchaggregator.md [hunk 1]
```diff
@@ -15,6 +15,9 @@ Properties
 ----------
 **fields:** Optional comma-separated list of fields to perform the distinct on. If not given, all fields are used.
 
+**numPartitions:** Number of partitions to user when grouping fields. If not specified, the execution
```
Comment (anwar6953) [inline core-plugins/docs/Distinct-batchaggregator.md]: s/user/use

File: core-plugins/docs/GroupByAggregate-batchaggregator.md [hunk 2]
```diff
@@ -29,6 +29,8 @@ The second will create a field called ``cheapest`` that contains the minimum ``p
 The count function differs from count(*) in that it contains non-null values of a specific field,
 while count(*) will count all records regardless of value.
 
+**numPartitions:** Number of partitions to user when grouping fields. If not specified, the execution
```
Comment (anwar6953) [inline core-plugins/docs/GroupByAggregate-batchaggregator.md]: s/user/use

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/AggregatorConfig.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import co.cask.cdap.api.plugin.PluginConfig;
+
+import javax.annotation.Nullable;
+
+/**
+ * Base aggregator config, since almost all aggregators should support setting the number of partitions.
+ */
+public class AggregatorConfig extends PluginConfig {
+  @Nullable
+  @Description("Number of partitions to use when aggregating. If none is specified, the execution framework " +
```
Comment (anwar6953) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/AggregatorConfig.java]: I think it sounds better:
`It not specified,`.
It's also then consistent with documentation in widget.json file.

**Comment by anwar6953** [comment]
3 minor comments, remainder LGTM

File: core-plugins/docs/Distinct-batchaggregator.md (hunk 1 above)
Comment (albertshau) [inline core-plugins/docs/Distinct-batchaggregator.md]: nice. this happens a lot to me for some reason. fingers autocorrect use to user.

**Pull request merged**